Code/Framework/AzToolsFramework/Tests/ActionManager/ActionManagerTests.cpp

```diff
@@ -107,6 +107,108 @@ namespace UnitTest
         EXPECT_TRUE(actionTriggered);
     }
 
+    TEST_F(ActionManagerFixture, GetActionName)
+    {
+        m_actionManagerInterface->RegisterActionContext("", "o3de.context.test", {}, m_widget);
+        
+        AzToolsFramework::ActionProperties actionProperties;
+        actionProperties.m_name = "Test Name";
+        m_actionManagerInterface->RegisterAction("o3de.context.test", "o3de.action.test", actionProperties, []{});
+
+        auto outcome = m_actionManagerInterface->GetActionName("o3de.action.test");
+        EXPECT_TRUE(outcome.IsSuccess());
+        EXPECT_EQ(outcome.GetValue(), "Test Name");
+    }
+
+    TEST_F(ActionManagerFixture, SetActionName)
+    {
+        m_actionManagerInterface->RegisterActionContext("", "o3de.context.test", {}, m_widget);
+        
+        AzToolsFramework::ActionProperties actionProperties;
+        actionProperties.m_name = "Wrong Name";
+        m_actionManagerInterface->RegisterAction("o3de.context.test", "o3de.action.test", actionProperties, []{});
+
+        auto setOutcome = m_actionManagerInterface->SetActionName("o3de.action.test", "Correct Name");
+        EXPECT_TRUE(setOutcome.IsSuccess());
+
+        auto getOutcome = m_actionManagerInterface->GetActionName("o3de.action.test");
+        EXPECT_EQ(getOutcome.GetValue(), "Correct Name");
+    }
+
+    TEST_F(ActionManagerFixture, GetActionDescription)
+    {
+        m_actionManagerInterface->RegisterActionContext("", "o3de.context.test", {}, m_widget);
+        
+        AzToolsFramework::ActionProperties actionProperties;
+        actionProperties.m_description = "Test Description";
+        m_actionManagerInterface->RegisterAction("o3de.context.test", "o3de.action.test", actionProperties, []{});
+
+        auto outcome = m_actionManagerInterface->GetActionDescription("o3de.action.test");
+        EXPECT_TRUE(outcome.IsSuccess());
+        EXPECT_EQ(outcome.GetValue(), "Test Description");
+    }
+
+    TEST_F(ActionManagerFixture, SetActionDescription)
+    {
+        m_actionManagerInterface->RegisterActionContext("", "o3de.context.test", {}, m_widget);
+        
+        AzToolsFramework::ActionProperties actionProperties;
+        actionProperties.m_description = "Wrong Description";
+        m_actionManagerInterface->RegisterAction("o3de.context.test", "o3de.action.test", actionProperties, []{});
+
+        auto setOutcome = m_actionManagerInterface->SetActionDescription("o3de.action.test", "Correct Description");
+        EXPECT_TRUE(setOutcome.IsSuccess());
```

Comment:
question: Is it worth also checking Get does return `"Correct Description"`?

Reply:
Look two lines below :D